packages/slice-machine/components/Simulator/index.tsx

```diff
@@ -114,17 +114,17 @@ const Simulator: ComponentWithSliceProps = ({ slice, variation }) => {
   const sharedSlice = useMemo(() => Slices.fromSM(slice.model), [slice.model]);
 
   const initialContent = useMemo<SharedSliceContent>(
-    () => slice.mock?.[0] || defaultSharedSliceContent(variation.id),
+    () =>
+      slice.mock?.find((m) => m.variation === variation.id) ||
+      defaultSharedSliceContent(variation.id),
     [slice.mock, variation.id]
   );
 
-  const previousInitialContent = useRef(initialContent);
   const [editorContent, setContent] = useState(initialContent);
 
-  if (previousInitialContent.current !== initialContent) {
-    previousInitialContent.current = initialContent;
+  useEffect(() => {
     setContent(initialContent);
-  }
+  }, [variation.id, initialContent]);
```

Comment:
hmmm,
so if variaion.id or initial content change, set change the ...editor content, got it :) (was worried that it might setting the initial content)

Reply:
Yea it’s only the editor. We keep track of initial content and api for the purpose of keeping slice I’d, hope we can simplify in the future